climada/util/event_yearset.py

```diff
@@ -0,0 +1,258 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Thu Dec  3 15:59:03 2020
+
+@author: carmensteinmann
+"""
+
+import copy
+import logging
+import numpy as np
+from numpy.random import default_rng
+
+from climada.util.dates_times import str_to_date
+
+LOGGER = logging.getLogger(__name__)
+
+def generate_yearset(eis, nr_resampled_years=None, year_list=None, multiple_events=True,
+                     sampling_vect=None, nr_events_per_year=None):
+
+    """PURPOSE:
+      convert an event (per occurrence) impact set (eis) into an annual impact
+      set (ais).
+
+    INPUTS:
+      eis (impact class object): an event impact set (eis)
+    OPTIONAL INPUT:
+        nr_resampled_years(int): the target number of years the impact yearset shall
+            contain.
+        year_list (list): list of years for the resulting annual impact set
+            (by default a list starting on the 01-01-0001 is generated)
+        multiple_events (boolean): True if the hazard causing the given impact can occur
+            several times per year (such as several tropical cyclones); False for hazards
+            that compute the impact on an annual scale (such as relative cropyield)
+        sampling_vect (array): the sampling vector, technical, see code (can be used to
+          re-create the exact same yearset). Needs to be obtained in a first
+          call, i.e. [ais,sampling_vect]=climada_eis2ais(...) and then
+          provided in subsequent calls(s) to obtain the exact same sampling
+          structure of yearset, i.e ais=climada_eis2ais(...,sampling_vect)
+        nr_events_per_year (array): amount of resampled events per year
+            (length = nr_resampled_years), can be reused similar to the sampling_vect
+            (to do: combine these two in one variable?)
+    OUTPUTS:
+      ais: the year impact set (ais), a struct with same fields as eis (such
+          as Value, ed, ...). All fields same content as in eis, except:
+          date(i): the year i
+          impact(i): the sum of impact for year(i).
+          frequency(i): the annual frequency, =1
+      sampling_vect: the sampling vector, technical, see code (can be used to
+                     re-create the exact same yearset)
+      """
+
+    # OLD CODE - STILL NEEDED?
+    # if years are known:
+    # events = list(EIS.calc_impact_year_set().keys())
+
+    # event_date = date_to_str(eis.date)
+    # event_year = [i.split('-', 1)[0] for i in event_date]
+
+    #n_years_his = np.ceil(1/min(eis.frequency)).astype('int')
+    # n_events = len(eis.event_id)
+
+
+    if not year_list:
+        year_list = [str(date) + '-01-01' for date in np.arange(1, nr_resampled_years+1).tolist()]
```

Comment:
I later compute the date in ordinal format (when saving it). This step allows the user to give a list of years as input - can it be left as is or would you suggest changing it?

Reply:
It is good yes! I added a comment because the variable `year_list` is now never used and thus can be removed.

https://github.com/CLIMADA-project/climada_python/pull/173/files#r604697615